include/swift/SIL/SILInstruction.h

```diff
@@ -1222,36 +1222,37 @@ struct SILDebugVariable {
 /// A DebugVariable where storage for the strings has been
 /// tail-allocated following the parent SILInstruction.
 class TailAllocatedDebugVariable {
+  using int_type = uint32_t;
   union {
-    uint32_t RawValue;
+    int_type RawValue;
     struct {
       /// Whether this is a debug variable at all.
-      unsigned HasValue : 1;
+      int_type HasValue : 1;
       /// True if this is a let-binding.
-      unsigned Constant : 1;
-      /// The source function argument position from left to right
-      /// starting with 1 or 0 if this is a local variable.
-      unsigned ArgNo : 16;
+      int_type Constant : 1;
       /// When this is nonzero there is a tail-allocated string storing
       /// variable name present. This typically only happens for
       /// instructions that were created from parsing SIL assembler.
-      unsigned NameLength : 14;
+      int_type NameLength : 14;
+      /// The source function argument position from left to right
+      /// starting with 1 or 0 if this is a local variable.
+      int_type ArgNo : 16;
     } Data;
-  };
+  } Bits;
```

Comment:
Why this change?

Reply:
Because the following is generated by the default warnings with clang/LLVM top-of-tree:
```
include/swift/SIL/SILInstruction.h:1227:5: warning: anonymous types declared in an anonymous union are an extension [-Wnested-anon-types]
    struct {
    ^
```